lib/src/inherited_provider.dart

```diff
@@ -602,23 +602,23 @@ class _CreateInheritedProvider<T> extends _Delegate<T> {
 
 @visibleForTesting
 // ignore: public_member_api_docs
-bool debugIsInInheritedProviderUpdate = false;
+bool? debugIsInInheritedProviderUpdate = false;
 
 @visibleForTesting
+bool? debugIsInInheritedProviderCreate = false;
 // ignore: public_member_api_docs
-bool debugIsInInheritedProviderCreate = false;
 
 class _CreateInheritedProviderState<T>
-    extends _DelegateState<T, _CreateInheritedProvider<T>> {
-  VoidCallback _removeListener;
+    extends _DelegateState<T?, _CreateInheritedProvider<T?>?> {
+  VoidCallback? _removeListener;
   bool _didInitValue = false;
-  T _value;
-  _CreateInheritedProvider<T> _previousWidget;
+  T? _value;
```

Comment:
`_didInitValue` made me think _value can be late initialized. I ran into problems primarily due to _create function being optional. _update takes in _value which then causes a read before a write.

Is null an acceptable `_value`? @rrousselGit